src/libs/api-gateway.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type { APIGatewayProxyEvent, APIGatewayProxyResult, Handler } from 'aws-lambda';
+import type { FromSchema } from 'json-schema-to-ts';
+import { ServerlessMysql } from 'serverless-mysql';
+
+type ValidatedAPIGatewayProxyEvent<S> = Omit<APIGatewayProxyEvent, 'body'>
```

Comment:
There are some reasons I kept it as a type.
1. The `aws-nodejs-typescript` [official template](https://www.serverless.com/framework/docs/providers/aws/cli-reference/create) generated this code like this
2. There is an ongoing debate on which is better to use: types or interfaces. There are some positive arguments about preferring types when not needing something specific for interfaces.
    1. [Relevant 4min video](https://www.youtube.com/watch?v=zM9UPcIyyhQ) from a developer that interacts with the Typescript team
4. I couldn't identify an objective reason for me to change it, other than my familiarity with interfaces

So for now I prefer an interface, but kept trusting the original template developer and didn't change it.

Reply:
In the video there is an argument for interfaces to be made.
IMO the core difference between interfaces and types are that interfaces describe the "shape" of an object and types are more closely related to structs for other languages.

This instance seems like an interface because the middleware receives a `APIGatewayProxyEvent` and passes a `ValidatedAPIGatewayProxyEvent` which inherits `APIGatewayProxyEvent` adding some properties, making a type not the most indicated.

Reply:
Trying to convert both these types to interfaces I ran into an issue with the [`json-schema-to-ts`](https://www.npmjs.com/package/json-schema-to-ts) dependency:
https://github.com/ThomasAribart/json-schema-to-ts/issues/32

Basically it converts a Schema ( example at [`functions/create/schema.ts`](https://github.com/HathorNetwork/hathor-atomic-swap-service/blob/1c938262d62236d1ba9af8634626f11fb5a14aec/src/functions/create/schema.ts) ) into a type and TypeScript is then able to validate it at compilation time.

Without it, Typescript would not be able to accept, for example, this cast type:
https://github.com/HathorNetwork/hathor-atomic-swap-service/blob/1b5302b4ed06a6ebf9553100981ae8803a6c5671/src/functions/create/handler.ts#L18-L21

It would be necessary to use `unknown` as an intermediary:
```ts
...
} = event.body as unknown as ICreateProposalRequest;
...
```

Do you agree that these definitions at the API Gateway and handlers would be better left as a `type`, even if it doesn't quite fit our existing code style?